src/api/console_stringifier.py

```diff
@@ -33,16 +34,61 @@ def _modify_unit(self, unit: str) -> str:
         """
         Returns the modified unit.
         """
-        unit = (
-            unit.replace(r"\squared", "^2")
-            .replace(r"\cubed", "^3")
-            .replace("\\per\\", "/")
-            .replace(r"\per", "/")
-            .replace("\\", " ")
-            .strip()
-        )
-
-        if unit[0] == "/":
-            unit = f"1{unit}"
+
+        # Detect "\squared" etc.:
+        unit = unit.replace(r"\squared", "^2").replace(r"\cubed", "^3")
+        unit = re.sub(r"(\s+)\^", "^", unit)
+
+        # Detect special units:
+        unit = unit.replace(r"\percent", r"\%").replace(r"\degree", "°")
```

Comment:
I'm still not sure why you replace `\percent` by `r"\%"` and not by `"%"`. The `\` will get split in the next step anyways, so why do you want to add it in the first place? This is not the reason why we get spaces around `%`, it's the `" ".join(...)` that does that.

Reply:
Without the `\`, the string is never split at the right location!

Reply:
Here is an actual test case: Consider the line `wiz.res("e", "1.0", r"\per\percent")`. With the current code, the output is as expected, namely `e = 1.0 1/%`. However, if we replace `\%` by just `%` in lune 43, we get instead: `e = 1.0 per%`; hence, a faulty output.

Reply:
Thanks for the example, I now get it ;) It's because we need this in the line `unit_parts = re.split(r"[\\|\s]+", unit)`. I've added your example to the playground such that we don't forget to add it to the unit tests later on.